### PR TITLE
Fixed #34212 -- Made RedisCacheClient.incr() use write connection.

### PR DESCRIPTION
**Pull request by select-case**

### CODE REVIEW

**Comment by felixxm** [comment]
@select-case Thanks :+1: Can you add a regression test? For example:
```diff
diff --git a/tests/cache/tests.py b/tests/cache/tests.py
index c4d4522514..8653ff10ff 100644
--- a/tests/cache/tests.py
+++ b/tests/cache/tests.py
@@ -1787,6 +1787,13 @@ class RedisCacheTests(BaseCacheTests, TestCase):
     def incr_decr_type_error(self):
         return self.lib.ResponseError
 
+    def test_incr_write_connection(self):
+        with mock.patch(
+            "django.core.cache.backends.redis.RedisCacheClient.get_client"
+        ) as mocked_get_client:
+            cache.incr("answer")
+            self.assertEqual(mocked_get_client.call_args.kwargs, {"write": True})
+
     def test_cache_client_class(self):
         self.assertIs(cache._class, RedisCacheClient)
         self.assertIsInstance(cache._cache, RedisCacheClient)

```

File: tests/cache/tests.py [hunk 1]
```diff
@@ -1786,9 +1786,11 @@ def setUp(self):
...
     def incr_decr_type_error(self):
         return self.lib.ResponseError
-    
+
     def test_incr_write_connection(self):
-        with mock.patch("django.core.cache.backends.redis.RedisCacheClient.get_client") as mocked_get_client:
+        with mock.patch(
+            "django.core.cache.backends.redis.RedisCacheClient.get_client"
+        ) as mocked_get_client:
             cache.incr("answer")
```
[the remark this answers is not in the document]
Comment (sinwoobang) [reply tests/cache/tests.py:1794]: Actually, I was working on this stuff samely. Would you mind if I patch it based on yours? @select-case

[the remark this answers is not in the document]
Comment (sinwoobang) [reply tests/cache/tests.py:1794]: Could you please review PR? https://github.com/select-case/django/pull/2 @select-case

**Comment by felixxm** [comment]
@select-case @sinwoobang Thanks both :+1:

**Pull request merged**